scripts/diagnostics-agent-setup.ps1

```diff
@@ -42,15 +42,14 @@ function New-DiagnosticsAgent {
     $diagnosticsBinary = Join-Path $DIAGNOSTICS_DIR "dcos-diagnostics.exe"
     $logFile = Join-Path $DIAGNOSTICS_LOG_DIR "diagnostics-agent.log"
     New-Item -ItemType File -Path $logFile
-
     $dcos_endpoint_config_dir = Join-Path $DIAGNOSTICS_DIR "dcos-diagnostics-endpoint-config.json"
-
     $diagnosticsAgentArguments = (  "daemon " + `
                                     "--role agent " + `
                                     "--debug " + `
                                     "--no-unix-socket " + `
                                     "--port `"${DIAGNOSTICS_AGENT_PORT}`" " + `
                                     "--endpoint-config `"${dcos_endpoint_config_dir}`"")
+    (Get-Service -Name "dcos-*").Name | Out-File -Encoding ascii -FilePath "${DIAGNOSTICS_DIR}\servicelist.txt"
```

Comment:
The original purpose of servicelist.txt is to provide a list of services that a DC/OS was intended to be setup with -- it's prepared before the deployment started. The proposed change here will change its original intention because in the case that a service setup (eg. diagnostics or  metrics services) completely failed, that is, not even added as one of the Windows service, this new proposed change would not catch such a deployment issue.

Reply:
I don't quite understand what issue are you pointing out there.

The only problem that I see with having a static `servicelist.txt` is that we can't handle both these scenarios:
- If DC/OS version is < `1.11`, then the monitored services are: `dcos-spartan`, `dcos-epmd` and `dcos-mesos-slave`
- If DC/OS version is >= `1.11`, then the monitored services are: `dcos-net` and `dcos-mesos-slave`.

We only know what DC/OS version we have during the setup and that's why I believe that `servicelist.txt` generation should be during the setup phase.